Client/Application/Delegates/AppDelegate.swift

```diff
@@ -344,6 +305,57 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
       
         return shouldPerformAdditionalDelegateHandling
     }
+    
+    func handleReferralLookup(_ urp: UserReferralProgram, checkClipboard: Bool) {
+        let initialOnboarding =
+            Preferences.General.basicOnboardingProgress.value == OnboardingProgress.none.rawValue
+        
+        // FIXME: Update to iOS14 clipboard api once ready (#2838)
+        if initialOnboarding && UIPasteboard.general.hasStrings {
+            log.debug("Skipping URP call at app launch, this is handled in privacy consent onboarding screen")
+            return
+        }
+        
+        if Preferences.URP.referralLookupOutstanding.value == true {
+            var refCode: String?
+            
+            let savedRefCode = checkClipboard ? UIPasteboard.general.string : nil
+            
+            if Preferences.URP.referralCode.value == nil {
```

Comment:
@iccub - May be move this check before `UIPaseboard.general.string`?

Reply:
It has to be here, we sanitize the clipboard output after it has been read 
`UserReferralProgram.sanitize(input: savedRefCode)`

Reply:
I will move this, but in theory it should never happen